include/Piece.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <iostream>
+#include <vector>
+using std::vector;
+#include <string>
+#include "Move.h"
+#include "Location.h"
+
+
+// Forward declaration of Board
+class Board;
+
+enum Color { WHITE, BLACK ,NONE};
+
+class Piece {
+protected:
+    Color color;
+    Location location;
+    char symbol = 'x';
+    Board* board;
+
+public:
+    Piece(Color color, Location location, Board* board);
+    virtual ~Piece();
+
+
+    Color getColor() const;
+    Location getLocation() const;
+    char getSymbol() const;
+    void move(Location endLocation);
+    virtual vector<Move> getAllLegalMoves() const = 0;    
```

Comment:
bad way for checking the moves, in each time you call the function, you make the same new calculations, you don't need that !!